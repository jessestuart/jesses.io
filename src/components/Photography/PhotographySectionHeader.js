import React from 'react'
import PropTypes from 'prop-types'
import Link from 'gatsby-link'
import { DateTime } from 'luxon'

import StyledLink from '../StyledLink/StyledLink'

const PhotographySectionHeader = ({ datetime, href }) => (
  <div className="alegreya-sans f2 fw5 header-primary mb2 tr w-100">
    <Link style={{ textDecoration: 'none' }} to={href}>
      <StyledLink className="db">
        {datetime.toFormat('dd LLLL yyyy')}
      </StyledLink>
    </Link>
  </div>
)

PhotographySectionHeader.propTypes = {
  datetime: PropTypes.instanceOf(DateTime).isRequired,
  href: PropTypes.string.isRequired,
}

export default PhotographySectionHeader
