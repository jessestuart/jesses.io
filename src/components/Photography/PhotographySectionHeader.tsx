import Link from 'gatsby-link'
import { DateTime } from 'luxon'
import React from 'react'
import { Text } from 'rebass/styled-components'

import Colors from 'utils/colors'

interface Props {
  datetime: DateTime
  href?: string
}

const PhotographySectionHeader = ({
  datetime,
  href = '/photography',
}: Props) => (
  <Text
    as="h2"
    borderBottom={`1px solid ${Colors.gray.calm}`}
    className="flex fw5"
    flex="1"
    fontSize="6"
    justifyContent="flex-end"
    marginBottom="2"
    width="100%"
    sx={{
      color: 'textMediumMuted',
      fontFamily: 'smallcaps',
    }}
  >
    <Link to={href}>{datetime.toFormat('dd LLLL yyyy')}</Link>
  </Text>
)

export default PhotographySectionHeader
