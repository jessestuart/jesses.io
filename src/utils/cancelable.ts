import Promise from 'bluebird'
import _ from 'lodash'

export default class Cancelable {
  public cancelled: boolean
  public promise: Promise<any>

  constructor(promise: Promise<any>) {
    this.cancelled = false
    this.promise = new Promise((resolve, reject) => {
      if (!this.cancelled) {
        return promise.then(resolve, reject)
      }
      return Promise.resolve()
    })
  }

  public then = (cb: () => any) =>
    this.cancelled ? Promise.resolve() : Promise.resolve(this.promise.then(cb))

  public cancel = () => _.set(this, 'cancelled', true)
}
