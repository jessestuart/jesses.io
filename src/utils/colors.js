import gray from 'gray-percentage'

const colors = {
  primary: {
    // Light Primary shades --
    // autogenerated between #e4e4e4 and #ff3a5c.
    light1: '#E6D2D6',
    light2: '#E9C1C8',
    light3: '#ECB0BB',
    light4: '#EE9FAD',
    light5: '#F18FA0',
    light6: '#F47D92',
    light7: '#F66D84',
    light8: '#F95B77',
    light9: '#FC4A69',
    // Main primary.
    main: '#ff3a5c',
    // Dark Primary shades --
    // autogenerated between #333 and #ff3a5c.
    dark9: '#EA3957',
    dark8: '#D63853',
    dark7: '#C1374F',
    dark6: '#AD374B',
    dark5: '#993647',
    dark4: '#843543',
    dark3: '#70353F',
    dark2: '#5B343B',
    dark1: '#473337',
  },
  secondary: {
    // Light Primary shades --
    // autogenerated between #e4e4e4 and     #673ab7.
    light1: '#d7d2df',
    light2: '#cac1da',
    light3: '#beb0d6',
    light4: '#b19fd1',
    light5: '#a58fcd',
    light6: '#987dc8',
    light7: '#8c6dc4',
    light8: '#7f5bbf',
    light9: '#734abb',
    main: '#673ab7',
  },
  bgDark: 'rgb(55, 59, 70)',
  accent: '',
  success: '',
  warning: '',
  ui: {
    bright: '#FFC9D2',
    light: '#f5f3f7',
    whisper: '#fbfafc',
  },
  gray: {
    dark: gray(8, 270),
    copy: gray(12, 270),
    calm: gray(46, 270),
  },
}

export default colors
