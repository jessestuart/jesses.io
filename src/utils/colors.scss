$gray-100: #F8F9FA !default;
$gray-200: #E9ECEF !default;
$gray-300: #DEE2E6 !default;
$gray-400: #CED4DA !default;
$gray-500: #ADB5BD !default;
$gray-600: #868E96 !default;
$gray-700: #495057 !default;
$gray-800: #343A40 !default;
$gray-900: #212529 !default;
$black: $gray-800 !default;
$white: $gray-400 !default;

$blue: #56E2FF !default;
$indigo: #3F51B5 !default;
$purple: #673AB7 !default;
// $pink: #FC718D !default;
$pink: #FF3A5C !default;
$red: #F44336 !default;
$orange: #FF5722 !default;
$yellow: #FBEBA9 !default;
$green: #B4DF75 !default;
$teal: #009688 !default;
$cyan: #00BCD4 !default;

$background: #343a40;
// $background: #282C34;

// Reassign color vars to semantic color scheme
$theme-colors: (
  primary: $purple,
  secondary: $gray-600,
  success: $green,
  info: $blue,
  warning: $yellow,
  danger: $red,
  light: $gray-300,
  dark: $gray-800,
  background: $background
) !default;
