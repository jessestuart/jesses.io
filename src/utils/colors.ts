import gray from 'gray-percentage'
import color from 'onecolor'

type ColorMap = { [colorname: string]: string }

interface Colors {
  accent: string
  bgDark: string
  defaultHover: string
  defaultLink: string
  gray: ColorMap
  primary: ColorMap
  secondary: ColorMap
  ui: ColorMap
}

const primaryColors = {
  // Light Primary shades.
  light1: '#ffebee',
  light2: '#ffd7de',
  light3: '#ffc3ce',
  light4: '#ffb0bd',
  light5: '#ff9cad',
  light6: '#ff889d',
  light7: '#ff758c',
  light8: '#ff617c',
  light9: '#ff4d6c',
  // Main primary.
  main: '#FF3A5C',
  // Dark Primary shades --
  // autogenerated between #000 and #ff3a5c.
  dark9: '#e53452',
  dark8: '#cc2e49',
  dark7: '#b22840',
  dark6: '#992237',
  dark5: '#7f1d2e',
  dark4: '#661724',
  dark3: '#4c111b',
  dark2: '#330b12',
  dark1: '#190509',
}

const secondaryColors = {
  // Light Secondary shades --
  light0: '#FBFAFC',
  light1: '#efebf7',
  light2: '#e0d7f0',
  light3: '#d1c3e9',
  light4: '#c2b0e2',
  light5: '#b39cdb',
  light6: '#a388d3',
  light7: '#9475cc',
  light8: '#8561c5',
  light9: '#764dbe',
  main: '#673ab7',
  dark9: '#5c34a4',
  dark8: '#522e92',
  dark7: '#482880',
  dark6: '#3d226d',
  dark5: '#331d5b',
  dark4: '#291749',
  dark3: '#1e1136',
  dark2: '#140b24',
  dark1: '#0a0512',
}

const colors: Colors = {
  accent: 'rgba(50, 0, 50, 0.8)',
  primary: primaryColors,
  secondary: secondaryColors,

  // blend: {
  //   blend1: '#773aac',
  //   blend2: '#8839a2',
  //   blend3: '#993998',
  //   blend4: '#aa3a8e',
  //   blend5: '#bb3984',
  //   blend6: '#cc3a7a',
  //   blend7: '#dd3a70',
  //   blend8: '#ee3a66',
  // },

  bgDark: 'rgb(55, 59, 70)',
  // headerDark: '#282C34',

  ui: {
    bright: '#FFC9D2',
    light: '#f5f3f7',
    whisper: '#fbfafc',
  },

  gray: {
    dark: gray(8, 270),
    copy: gray(12, 270),
    calm: gray(46, 270),
  },

  defaultHover: color(secondaryColors.light7)
    .alpha(0.8)
    .cssa(),

  defaultLink: color(primaryColors.main)
    .alpha(0.8)
    .cssa(),
}

export default colors
