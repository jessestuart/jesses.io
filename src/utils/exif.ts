import _ from 'lodash'

export const mapCameraModelExif = _.memoize(
  (model: string): string => {
    return model === 'ILCE-7M3' ? 'Sony α7 III' : model
  },
)

export const mapLensModelExif = _.memoize(
  (model: string): string => {
    const mapping = {
      '---': 'Rokinon 14mm ƒ2.8',
      'E 28-75mm F2.8-2.8': 'Tamron 28-75mm ƒ/2.8',
      'FE 85mm F1.8': 'Sony FE 85mm ƒ/1.8',
    }
    return mapping[model] || model
  },
)
