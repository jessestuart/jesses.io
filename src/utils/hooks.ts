import { graphql, useStaticQuery } from 'gatsby'

export const useSiteMetadata = () => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `,
  )
  return site.siteMetadata
}
