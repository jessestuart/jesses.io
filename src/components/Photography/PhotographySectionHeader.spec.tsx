import { DateTime } from 'luxon'
import React from 'react'

import { renderWithTheme } from '../../../test/utils/helpers'

import PhotographySectionHeader from 'components/Photography/PhotographySectionHeader'

jest.mock('react-headroom', () => 'Headroom')

const datetime = DateTime.utc(2019, 5, 1, 1, 1, 1)

test('Render PhotographySectionHeader.', () => {
  const component = renderWithTheme(
    <PhotographySectionHeader datetime={datetime} />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
