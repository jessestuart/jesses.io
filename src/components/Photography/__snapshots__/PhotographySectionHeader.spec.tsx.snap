// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PhotographySectionHeader. 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.7);
  font-family: Alegreya Sans SC;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 36px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 8px;
  width: 100%;
}

<h2
  className="c0 flex fw5"
  fontSize="6"
  width="100%"
>
  <a
    href="/true/photography"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    01 May 2019
  </a>
</h2>
`;
